src/sagemaker/model.py

```diff
@@ -236,10 +244,27 @@ def _compilation_job_config(
         }
         role = self.sagemaker_session.expand_role(role)
         output_model_config = {
-            "TargetDevice": target_instance_type,
             "S3OutputLocation": output_path,
         }
 
+        if target_instance_type is not None:
+            output_model_config["TargetDevice"] = target_instance_type
+        else:
+            assert target_platform_os is not None and target_platform_arch is not None, (
+                "target_instance_type or (target_platform_os and target_platform_arch) "
+                "should be provided"
+            )
```

Comment:
it's better practice to avoid using `assert`s in source code, and instead raise a `ValueError` in this case

https://stackoverflow.com/a/945135
https://stackoverflow.com/a/1838411

Reply:
Replaced with `ValueError`